.circleci/config.yml

```diff
@@ -768,37 +802,10 @@ workflows:
           # todo? add a link to the pipeline in the message
           message: ":runner: *Nightly run started!*"
       # check that `pip install wandb` is successful
-      - check_pip_install_wandb:
-          name: "pip-install-wandb-3.6"
-          image: "python:3.6"
-          context: slack-secrets
-          notify_on_failure: << pipeline.parameters.manual_nightly_slack_notify >>
-          requires:
-            - "slack-notify-on-start"
-      - check_pip_install_wandb:
-          name: "pip-install-wandb-3.7"
-          image: "python:3.7"
-          context: slack-secrets
-          notify_on_failure: << pipeline.parameters.manual_nightly_slack_notify >>
-          requires:
-            - "slack-notify-on-start"
-      - check_pip_install_wandb:
-          name: "pip-install-wandb-3.8"
-          image: "python:3.8"
-          context: slack-secrets
-          notify_on_failure: << pipeline.parameters.manual_nightly_slack_notify >>
-          requires:
-            - "slack-notify-on-start"
-      - check_pip_install_wandb:
-          name: "pip-install-wandb-3.9"
-          image: "python:3.9"
-          context: slack-secrets
-          notify_on_failure: << pipeline.parameters.manual_nightly_slack_notify >>
-          requires:
-            - "slack-notify-on-start"
-      - check_pip_install_wandb:
-          name: "pip-install-wandb-3.10"
-          image: "python:3.10"
+      - pip_install_wandb:
+          matrix:
+            parameters:
+              python_version: [ "3.6", "3.7", "3.8", "3.9", "3.10" ]
```

Comment:
many thanks @kptkin for the idea, it is waaay nicer this way! we should probably put all sorts of other things on the matrix, but not in this PR.

Reply:
Yeah agreed our configs could use some love... 
I wonder what is the feature parity between github and circleci since there more interesting configs we can use :)